src/melee.cpp

```diff
@@ -1940,3 +1945,129 @@ double player::unarmed_value() const
     // TODO: Martial arts
     return melee_value( ret_null );
 }
+
+void player::disarm( npc &target )
+{
+    if( !target.is_armed() ) {
+        return;
+    }
+
+    item &it = target.weapon;
+
+    const char *target_name = target.name.c_str();
+    const char *item_name = it.tname().c_str();
```

Comment:
const char *target_name = std::string(target.name).c_str();
    const char *item_name = std::string(it.tname()).c_str();

Would this be correct?